lib/warden/jwt_auth/strategy.rb

```diff
@@ -8,7 +8,9 @@ module JWTAuth
     # `Authorization` request header
     class Strategy < Warden::Strategies::Base
       def valid?
-        !token.nil?
+        issuer_claim_configured? ? issuer_claim_matches? : token_exists?
```

Comment:
`issuer_claim_matches?` is also dependant on the token being there (`token_exists?`). What about something like the following to make it clearer?

```ruby
token_exists? && issuer_claim_valid?

def issuer_claim_valid?
  configured_issuer = Warden::JWTAuth.config.issuer
  return true if configured_issuer.nil?
  
  payload = TokenDecoder.new.call(token)
  PayloadUserHelper.issuer_matches?(payload, configured_issuer)
end
```

Also, I'm a bit concerned as now we are calling twice `TokenDecoder`. One here and the other when `authenticate!` is invoked (through `UserDecoder`). Thinking out loud, but maybe we could change `UserDecoder` to take the payload instead of the token 🤔

Reply:
That was the part I was most unhappy with :D I like your suggestions, I've added it in 054fccd82dbc6cba414dad63c4ab1865244a877e

Since token parsing is not very expensive and basically can't vary in outcome, I am not too concerned about doing it twice. However, I agree that it feels cleaner if the UserDecover would get a parsed token already.